solidity/dashboard/src/services/token-staking.service.js

```diff
@@ -218,12 +219,15 @@ export const getOperatorsOfOwner = async (owner) => {
 
   // Fetch `StakeOwnershipTransferred` by operator field. We need to check more recent event
   // to make sure the delegation ownership has not been transferred.
-  const transferEventsByOperators = (
-    await stakingContract.getPastEvents("StakeOwnershipTransferred", {
-      fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingContract,
-      filter: { operator: operators },
-    })
-  ).reduce(reduceByOperator, {})
+  let transferEventsByOperators = {}
+  if (!isEmptyArray(operators)) {
+    transferEventsByOperators = (
+      await stakingContract.getPastEvents("StakeOwnershipTransferred", {
+        fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER.stakingContract,
+        filter: { operator: operators },
+      })
+    ).reduce(reduceByOperator, {})
```

Comment:
I need help to understand it. Say we have `StakeOwnershipTransferred(operator = X, newOwner = A)` and then `StakeOwnershipTransferred(operator = X, newOwner = B)`. What we are doing here is that we are eliminating `newOwner = A` event, correct?

Reply:
yep, exactly. I'll try to add more color here in a moment

Reply:
As a first step, we scan `StakeDelegated` by `owner` to get all operators. We can't 100% sure this relationship is still active since the owner can be changed.

Next, we need to scan `StakeOwnershipTransferred` by `newOwner` because another owner may have changed `owner` of a delegation to us. But again we can't 100% sure this relationship is still active since the owner can be changed.

So as the last step, we scan `StakeOwnershipTransferred` by all operators from these two results ^ and check the most recent `StakeOwnershipTransferred` event for a given operator to make sure we are the owner of the delegation.